go/libraries/doltcore/sqle/dprocedures/dolt_verify_constraints.go

```diff
@@ -75,37 +77,110 @@ func doDoltConstraintsVerify(ctx *sql.Context, args []string) (int, error) {
 		}
 	}
 
-	tableSet := set.NewStrSet(nil)
-	for _, val := range apr.Args {
-		_, tableName, ok, err := workingRoot.GetTableInsensitive(ctx, val)
-		if err != nil {
-			return 1, err
-		}
-		if !ok {
-			return 1, sql.ErrTableNotFound.New(tableName)
-		}
-		tableSet.Add(tableName)
+	tableSet, err := parseTablesToCheck(ctx, workingRoot, apr)
+	if err != nil {
+		return 1, err
 	}
 
-	newRoot, tablesWithViolations, err := merge.AddForeignKeyViolations(ctx, workingRoot, comparingRoot, tableSet, h)
+	// Check for all non-FK constraint violations
+	newRoot, tablesWithViolations, err := calculateViolations(ctx, workingRoot, comparingRoot, tableSet)
 	if err != nil {
 		return 1, err
 	}
 
+	if !outputOnly {
+		err = dSess.SetRoot(ctx, dbName, newRoot)
+		if err != nil {
+			return 1, err
+		}
+	}
+
 	if tablesWithViolations.Size() == 0 {
 		// no violations were found
 		return 0, nil
 	}
 
-	// violations were found
+	// TODO: We only return 1 or 0 to indicate if there were any constraint violations or not. This isn't
+	//       super useful to customers, and not how the CLI command works. It would be better to return
+	//       results that indicate the total number of violations found for the specified tables, and
+	//       potentially also a human readable message.
+	return 1, nil
+}
 
-	if !outputOnly {
-		err = dSess.SetRoot(ctx, dbName, newRoot)
+// calculateViolations calculates all constraint violations between |workingRoot| and |comparingRoot| for the
+// tables in |tableSet|. Returns the new root with the violations, and a set of table names that have violations.
+// Note that constraint violations detected for ALL existing tables will be stored in the dolt_constraint_violations
+// tables, but the returned set of table names will be a subset of |tableSet|.
+func calculateViolations(ctx *sql.Context, workingRoot, comparingRoot *doltdb.RootValue, tableSet *set.StrSet) (*doltdb.RootValue, *set.StrSet, error) {
```

Comment:
This just seems like an odd way to structure this process. Naively I would expect that constraint violation checks are their own library methods that are used by both the merge code as necessary, as well as here (where we do a full table scan).

Reply:
We don't always do a full table scan though. The default mode of `dolt_verify_constraints()` only checks changes in the working set, so at some point, we have to support diffing HEAD with the working set and only verifying those changes.